core/arch/asm_defines.asm

```diff
@@ -458,6 +498,25 @@ ASSUME fs:_DATA @N@\
 #  define ARG8_NORETADDR  DWORD [7*ARG_SZ + esp]
 #  define ARG9_NORETADDR  DWORD [8*ARG_SZ + esp]
 #  define ARG10_NORETADDR DWORD [9*ARG_SZ + esp]
+#  ifdef WINDOWS
+#  define PUSH_CALLEE_SAVED_REGS() \
+        PUSH_SEH(REG_XBX) @N@\
+        PUSH_SEH(REG_XBP) @N@\
+        PUSH_SEH(REG_XSI) @N@\
+        PUSH_SEH(REG_XDI)
+#  define POP_CALLEE_SAVED_REGS() \
+        POP_SEH(REG_XDI) @N@\
+        POP_SEH(REG_XSI) @N@\
+        POP_SEH(REG_XBP) @N@\
+        POP_SEH(REG_XBX)
+#  else /* UNIX */
+#   define PUSH_CALLEE_SAVED_REGS() \
+         PUSH_SEH(REG_XBX) @N@\
```

Comment:
UNIX should match Windows 32-bit: cdecl is all the same, with ebx, esi, edi, and ebp callee-saved, AFAIK.  Where are you seeing that esi and edi are caller-saved?

Reply:
I googled multiple places, one is https://en.wikipedia.org/wiki/X86_calling_conventions
They all said System V ABI does not declare xdi, xsi callee-saved. I'll check if I find a better place of documentation

Reply:
That wiki page says that for the System V **AMD64** ABI!  We're talking about 32-bit here.

Reply:
You're right! I was making the assumption that i386 is the same. https://wiki.osdev.org/System_V_ABI#i386.
I'll fix